src/node/mod.rs

```diff
@@ -0,0 +1,1241 @@
+
+        self.stage = Stage::Bootstrapping(stage);
+    }
+
+    // Transition from Joining to Bootstrapping on join failure
+    fn rebootstrap(&mut self, network_cfg: NetworkParams) {
+        // TODO: preserve relocation details
+        self.stage = Stage::Bootstrapping(Bootstrapping::new(network_cfg, None));
+        self.core.full_id = FullId::gen(&mut self.core.rng);
+        self.core.transport.bootstrap();
+    }
+
+    fn set_log_ident(&self) -> log_utils::Guard {
+        use std::fmt::Write;
+        log_utils::set_ident(|buffer| match &self.stage {
+            Stage::Bootstrapping(_) => write!(buffer, "Bootstrapping({}) ", self.name()),
+            Stage::Joining(stage) => write!(
+                buffer,
+                "Joining({}({:b})) ",
+                self.name(),
+                stage.target_section_prefix()
+            ),
+            Stage::Approved(stage) if !stage.chain.is_self_elder() => write!(
+                buffer,
+                "Adult({}({:b})) ",
+                self.core.name(),
+                stage.chain.our_prefix()
+            ),
+            Stage::Approved(stage) => write!(
+                buffer,
+                "Elder({}({:b})) ",
+                self.core.name(),
+                stage.chain.our_prefix()
+            ),
+            Stage::Terminated => write!(buffer, "Terminated"),
+        })
+    }
+}
+
+#[cfg(feature = "mock_base")]
+impl Node {
+    /// Returns whether the node is approved member of a section.
+    pub fn is_approved(&self) -> bool {
+        self.stage.approved().is_some()
+    }
+
+    /// Returns whether the node is Elder.
+    pub fn is_elder(&self) -> bool {
+        self.chain()
+            .map(|chain| chain.is_self_elder())
+            .unwrap_or(false)
+    }
+
+    /// Indicates if there are any pending observations in the parsec object
+    pub fn has_unpolled_observations(&self) -> bool {
+        self.stage
+            .approved()
+            .map(|stage| stage.parsec_map.has_unpolled_observations())
+            .unwrap_or(false)
+    }
+
+    /// Indicates if there are any pending observations in the parsec object
+    pub fn unpolled_observations_string(&self) -> String {
+        self.stage
+            .approved()
+            .map(|stage| stage.parsec_map.unpolled_observations_string())
+            .unwrap_or_else(String::new)
+    }
+
+    /// Returns whether the given peer is an elder of our section.
+    pub fn is_peer_our_elder(&self, pub_id: &PublicId) -> bool {
+        self.stage
+            .approved()
+            .map(|stage| stage.chain.is_peer_our_elder(pub_id))
+            .unwrap_or(false)
+    }
+
+    /// Send a message to the given targets using the given delivery group size.
+    pub fn send_message_to_targets(
+        &mut self,
+        dst_targets: &[SocketAddr],
+        dg_size: usize,
+        message: Message,
+    ) -> Result<(), RoutingError> {
+        let message = message.to_bytes()?;
+        self.core
```

Comment:
Ok(self.core
            .send_message_to_targets(dst_targets, dg_size, message))
?

Reply:
Clippy doesn't like that: https://rust-lang.github.io/rust-clippy/master/index.html#unit_arg